src/main/java/org/opensearchmetrics/metrics/release/CodeCoverage.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearchmetrics.metrics.release;
+
+import org.apache.http.util.EntityUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.opensearchmetrics.model.codecov.CodeCovResponse;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Optional;
+
+public class CodeCoverage {
+    private final CloseableHttpClient httpClient;
+
+    @Inject
+    public CodeCoverage() {
+        this(HttpClients.createDefault());
+    }
+    CodeCoverage(CloseableHttpClient httpClient) {
```

Comment:
NIT: add @VisibleForTesting

Reply:
Do we need `@VisibleForTesting` here?

Reply:
It's not needed, but a good to have since it alerts the user when they attempt to use that constructor signature outside of testing